### PR TITLE
263 smtp implementation

### PR DESCRIPTION
**Pull request by sethalt**

### CODE REVIEW

File: server/email-util.es6 [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    to: to,
+    subject: subject,
+    text: body // plain text
+  };
+
+  let trans = transporter.sendMail(mailOptions, (error, info) => {
+    if (error) {
+      return console.log('error message', error);
+    }
+    console.log('message info', info);
```
Comment (dubharmonic) [inline server/email-util.es6]: can we remove the console.log?

File: server/vcap-services.es6 [hunk 2]
```diff
@@ -52,3 +52,13 @@ vcapConstants.eAuthCert = eAuthService.credentials.cert;
...
 module.exports = vcapConstants;
+
+// SMTP
+let smtp = VCAPServices['user-provided'].find(element => {
+  return element.name === 'smtp-service';
+});
+vcapConstants.smtpHost = smtp.credentials.smtpserver;
+vcapConstants.smtpUsername = smtp.credentials.username;
+if (smtp.credentials.password) {
+  vcapConstants.smtpPassword = smtp.credentials.password;
```
Comment (dubharmonic) [inline server/vcap-services.es6]: if there's no password defined, smtpPassword will be undefined with or without the if statement, right?

Comment (sethalt) [reply server/vcap-services.es6]: Correct and updated

File: server/email-util.es6 (hunk 1 above)
Comment (sethalt) [reply server/email-util.es6]: Yep, removed

**Pull request merged**